From 0b8ca9cc91b73003ca580af08f4281239a92ba65 Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@cloudera.com>
Date: Wed, 22 Aug 2012 12:01:24 -0700
Subject: [PATCH 1176/1357] MR1: CDH-7412. Speedup counter name resolution during ResourceBundle lookup
 	  Backport of MR-2855(trunk) and MR-4565(branch-1)

Reason: Counter name resolution taking too long
Ref: CDH-7412
Author: Karthik Kambatla
---
 .../hadoop/mapreduce/util/ResourceBundles.java     |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/mapred/org/apache/hadoop/mapreduce/util/ResourceBundles.java b/src/mapred/org/apache/hadoop/mapreduce/util/ResourceBundles.java
index aede782..52addcf 100644
--- a/src/mapred/org/apache/hadoop/mapreduce/util/ResourceBundles.java
+++ b/src/mapred/org/apache/hadoop/mapreduce/util/ResourceBundles.java
@@ -18,6 +18,7 @@
 
 package org.apache.hadoop.mapreduce.util;
 
+import java.util.Locale;
 import java.util.ResourceBundle;
 import java.util.MissingResourceException;
 
@@ -33,7 +34,8 @@ public class ResourceBundles {
    * @throws MissingResourceException
    */
   public static ResourceBundle getBundle(String bundleName) {
-    return ResourceBundle.getBundle(bundleName.replace('$', '_'));
+    return ResourceBundle.getBundle(bundleName.replace('$', '_'),
+        Locale.getDefault(), Thread.currentThread().getContextClassLoader());
   }
 
   /**
-- 
1.7.0.4

